app/src/main/java/org/schabi/newpipe/util/TextLinkifier.java

```diff
@@ -30,92 +39,173 @@ private TextLinkifier() {
+     * StreamingService, String)}
      * after creating an {@link Markwon} object and using
      * {@link Markwon#setMarkdown(TextView, String)}.
      *
-     * @param context       the context to use
-     * @param markdownBlock the block of markdown text to be linked
-     * @param textView      the TextView to set the plain text block linked
+     * @param context          the context to use
+     * @param markdownBlock    the block of markdown text to be linked
+     * @param textView         the TextView to set the plain text block linked
+     * @param streamingService the {@link StreamingService} of the content
+     * @param contentUrl       the URL of the content
      * @return a disposable to be stored somewhere and disposed when activity/fragment is destroyed
      */
     public static Disposable createLinksFromMarkdownText(final Context context,
                                                          final String markdownBlock,
-                                                         final TextView textView) {
+                                                         final TextView textView,
+                                                         final StreamingService streamingService,
+                                                         final String contentUrl) {
         final Markwon markwon = Markwon.builder(context).usePlugin(LinkifyPlugin.create()).build();
         markwon.setMarkdown(textView, markdownBlock);
-        return changeIntentsOfDescriptionLinks(context, textView.getText(), textView);
+        return changeIntentsOfDescriptionLinks(context, textView.getText(), textView,
+                streamingService, contentUrl);
+    }
+
+    /**
+     * Add click listeners which opens the popup player on timestamps in a plain text.
+     * <p>
+     * This method finds all timestamps in the {@link SpannableStringBuilder} of the description
+     * using a regular expression, adds for each a {@link ClickableSpan} which opens the popup
+     * player at the time indicated in the timestamps.
+     *
+     * @param context              the context to use
+     * @param spannableDescription the SpannableStringBuilder with the text of the
+     *                             content description
+     * @param contentUrl           the URL of the content
+     * @param streamingService     the {@link StreamingService} of the content
+     */
+    private static void addClickListenersOnTimestamps(final Context context,
+                                                      final SpannableStringBuilder
+                                                              spannableDescription,
+                                                      final String contentUrl,
+                                                      final StreamingService streamingService) {
+        final String descriptionText = spannableDescription.toString();
```

Comment:
Are you making sure that the timestamp is not inside an url or attached to a word?

Reply:
No, I asked myself the question. Do you have a better regular expression?

Reply:
Adding `(?:^|\s)` at the beginning of the regex and `(?:\s|$)` at the end should do it, but do some checks [here](https://regex101.com/) to make sure.

Reply:
@Stypox Timestamps are successfully parsed, but results include spaces and linebreaks:

![Screenshot](https://user-images.githubusercontent.com/74829229/111906693-189fa880-8a52-11eb-98c3-c870cf1d8396.jpg)

Is it normal or a better regex is needed?

Reply:
That's normal, if you look at the capturing group content you will see it only contains the timestamp, but the regex also matches the spaces since it makes sure there is nothing at the beginning nor at the end.

Reply:
So, I need to remove spaces when adding click listeners or not?

Reply:
Yes, you need to remove spaces, I think

Reply:
I found better:
```java
            final int timestampStart = timestampMatches.start(2);
            final int timestampEnd = timestampMatches.end(3);
```